plugins/thunderbolt/fu-plugin-thunderbolt.c

```
@@ -744,6 +756,37 @@ fu_plugin_update (FuPlugin *plugin,
 		return FALSE;
 	}
 
+	validation = fu_plugin_thunderbolt_validate_firmware (udevice,
+							      blob_fw,
+							      &error_local);
+	if (validation != VALIDATION_PASSED) {
+		g_autofree gchar* msg = NULL;
+		switch (validation) {
+		case VALIDATION_FAILED:
+			msg = g_strdup_printf ("could not validate firmware: %s",
+					       error_local->message);
+			break;
+		case VALIDATION_PASSED_UNKNOWN_DEVICE:
+			msg = g_strdup ("firmware validation seems to be passed but the device is unknown");
+			break;
+		default:
+			break;
+		}
+		if (!force) {
+			g_set_error (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INVALID_FILE,
+				     "%s\n"
+				     "To update anyway please run the update with --force.",
+				     msg);
+			return FALSE;
+		}
+		g_warning ("%s\n"
```

again, maybe "ignoring error as forcing: %s", msg)